app/src/main/java/com/alexbar/layoutbasic/screen/confirm_payment/widgets/AddNewCreditCardDialog.kt

```diff
@@ -0,0 +1,121 @@
+package com.alexbar.layoutbasic.screen.confirm_payment.widgets
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.*
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.window.Dialog
+import com.alexbar.layoutbasic.ui.theme.Dimens.dimen_16_dp
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_cancel
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_full_name
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_month
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_number
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_save
+import com.alexbar.layoutbasic.utils.ConfirmPayment.add_credit_card_year
+
+@Composable
+fun AddCreditCardDialog(
+    onDismissRequest: () -> Unit
+) {
+    var name by remember { mutableStateOf("") }
+    var number by remember { mutableStateOf("") }
+    var expirationMonth by remember { mutableStateOf("") }
+    var expirationYear by remember { mutableStateOf("") }
+
+    Dialog(onDismissRequest = onDismissRequest) {
+        Box(modifier = Modifier
+                .fillMaxSize()
+                .background(Color.Black.copy(alpha = 0.5f))
+        )  {
```

Comment:
While creating my custom dialog, I realized it wasn't applying a dark overlay to the background screen. I attempted to implement it using this code, but I'm not sure if there's a more efficient approach. Also, when I launch the app, the dark overlay doesn't extend across the entire screen. I'm unsure whether this issue stems from a modifier that might have been adjusted in another part of the code, impacting this component, or if it's related to the technique I used to add the dark overlay. 

![Screenshot 2024-01-23 at 2 15 16 a m](https://github.com/em4n0101/LayoutBasic/assets/66237604/753216b9-c0d0-4549-84b2-fa2aa4c0cee6)

Reply:
for this matter i changed your code like this :
```
        Box(modifier = Modifier
            .clip(RoundedCornerShape(16.dp))
            .pointerInput(Unit) { detectTapGestures { } }
            .shadow(8.dp, shape = RoundedCornerShape(16.dp))
            .background(
                MaterialTheme.colorScheme.surface,
            ),
            contentAlignment = Alignment.Center
        ) 
```

![Uploading Screenshot 2024-01-26 at 10.18.55.png…]()